posthog/version_requirement.py

```diff
@@ -0,0 +1,68 @@
+from semantic_version.base import SimpleSpec, Version
+
+
+class VersionRequirement:
+    accepted_services = ("clickhouse", "postgres")
+
+    def __init__(self, service, version_range):
+        if service not in self.accepted_services:
+            services_str = ", ".join(self.accepted_services)
+            raise Exception(
+                f"service {service} cannot be used to specify a version requirement. service should be one of {services_str}"
+            )
+
+        self.service = service
+
+        try:
+            self.version_range = SimpleSpec(version_range)
+        except:
+            raise Exception(
+                f"version_range is invalid. See the Docs for SimpleSpec: https://pypi.org/project/semantic-version/"
+            )
+
+    def is_service_in_accepted_version(self):
+        service_version = self.get_service_version()
+        return [service_version in self.version_range, service_version]
+
+    def get_service_version(self):
+        if self.service == "postgres":
+            return self.get_postgres_version()
+
+        if self.service == "clickhouse":
+            return self.get_clickhouse_version()
+
+    def get_postgres_version(self):
+        from django.db import connection
+
+        with connection.cursor() as cursor:
+            cursor.execute("SHOW server_version")
+
+            rows = cursor.fetchone()
+            version = rows[0]
+
+        return self.version_string_to_semver(version)
+
+    def get_clickhouse_version(self):
+        from ee.clickhouse.client import sync_execute
+
+        rows = sync_execute("SELECT version()")
+        version = rows[0][0]
+
+        return self.version_string_to_semver(version)
+
+    @staticmethod
+    def version_string_to_semver(version):
```

Comment:
Shouldn't we use something like:

```
>>> import semantic_version
>>> v = semantic_version.Version('0.1.1')
>>> v.major
0
```
?

Reply:
That's what I had, but that won't parse a lot of stuff we want to parse. 

e.g. my local Postgres is on version `14.0`, which will throw here.

ClickHouse also has 4 levels in their versioning, which isn't supported either.

Hence the parsing. We lose a bit of granularity, but it's enough for us.

Reply:
I've added more handling. There are a few more edge cases, so this custom handling actually makes some sense.

e.g. `11.13 (Ubuntu 11.13-2.heroku1+1)`